docs/config/@config.md

```diff
@@ -0,0 +1,197 @@
+`@Config`
+=======
+
+The `@Config` annotation is an alternative to `Configuration`. 
+
+Table of Contents
+-------------
+
+* [Basics][basics]
+* [@Config Use][configuse]
+* [@Comment Use][commentuse]
+* [@Name Use][nameuse]
+* [@RangeInt Use][rangeintuse]
+* [@RangeDouble Use][rangedoubleuse]
+* [@LangKey Use][langkeyuse]
+* [@RequiresMcRestart Use][requiresmcrestartuse]
+* [@RequiresWorldRestart Use][requiresworldrestartuse]
+
+Basics
+------
+
+A Config should contain static fields, one per config option. There are a few exceptions, such as [Nested Configs][nestedconfigs] and [Sub Categories][subcategories]. Note that you can use an enum as a config field. To add information to and control config values, you can use the plethora of annotations provided in the `@Config` class.
+
+A good example of this system is [`Forge's Test`][forgetest]
+
+@Config Use
+-----------
+
+This annotation is used to denote a class is the container for some configuration options. There are 4 properties:
+
+|     Property|Type     | Default Value | Comment 
+|           -:|:-       |     :-:       |:-
+|        modid|`String` |     N/A       | The mod id that this configuration is associated with. 
+|         name|`String` |     `""`      | A user friendly name for the config file, the default will be modid. 
+|         type|`Type`   |`Type.INSTANCE`| The type this is, right now the only value is `Type.INSTANCE`. This is intended to be expanded upon later for more Forge controlled configs. 
+|     category|`String` |  `"general"`  | Root element category, defaults to "general", if this is an empty string then the root category is disabled. Any primitive fields will cause an error, and you must specify sub-category objects. 
+
+### Example
+```java
+@Config(modid = "reallycoolmod")
+public class Configs {
+
+}
+```
+
+@Comment Use
+------------
+
+Adding a `@Comment` annotation to a field will add a comment to the config in it's file. It has 1 property:
+
+| Property|Type       | Default Value |Comment 
+|       -:|:-         |       :-:     |:-
+|    value|`String[]` |       N/A     |This can be passed as a `String` or `String[]`, a `String[]` will form a multi-line comment where 1 element = 1 line.
+
+### Example
+```java
+@Comment({
+  "You can add comments using this",
+  "and if you supply an array it will me multiline"
+})
+public static boolean doTheThing = true;
+```
+This will produce a `boolean` config value with name `doTheThing`, the defualt value `true` and the comment provided in the config file.
```

Comment:
This is a really wordy sentence. I'm not sure how I'd reword it, but I think it could use some work.